lib/gears/timer.lua

```diff
@@ -91,7 +91,8 @@ function timer:start()
         gdebug.print_error(traceback("timer already started"))
         return
     end
-    self.data.source_id = glib.timeout_add(glib.PRIORITY_DEFAULT, self.data.timeout * 1000, function()
+    local timeout_ms = math.floor(self.data.timeout * 1000 + 0.5)
```

Comment:
Why do you need to add `+ 0.5`? Shouldn't you use `math.ceil` instead?

Reply:
`floor(x + 0.5) == round(x)`. Since there is no `math.round()` in Lua, this is the next best thing.

Reply:
oh, btw, we've got `gears.math.round()`

Reply:
The precision of flooring/ceiling/rounding milliseconds does not seem like it matters that much, I did the + 0.5 for rounding just out of perfectionism.

Changed that to `gmath.round` too